### PR TITLE
fix(button): colors update

### PR DESCRIPTION
**Pull request by theJohnnyMe**
**Describe pull-request**  
Buttons color update

**Solving issue**  
The button had some hard coded values for variations of black and white color with different opacity.
New color tokens are added as they are used in multiple places. 

Fixes: [CDEP-2999](https://tegel.atlassian.net/browse/CDEP-2999)

**How to test**  
1. Go to button component
2. Check colors for ghost, secondary and danger button in both dark and light mode
3. Check disabled, hover and active (pressed) state
4. Compare it with Figma
5. Color values need to match

**Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

[CDEP-2999]: https://tegel.atlassian.net/browse/CDEP-2999?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by alfredskeden — APPROVED** [review]
checked out code and looks good to me in all the states. LGTM

File: packages/core/src/components/button/button-vars.scss [hunk 1]
```diff
@@ -27,25 +27,24 @@
...
   --tds-btn-secondary-background-disabled: transparent;
-  --tds-btn-secondary-color-disabled: rgb(0 0 0 / 38%);
-  --tds-btn-secondary-border-color-disabled: rgb(0 0 0 / 38%);
+  --tds-btn-secondary-color-disabled: var(--tds-black-38);
+  --tds-btn-secondary-border-color-disabled: var(--tds-black-38);
   --tds-btn-icon-secondary-color-focus: var(--tds-black);
   --tds-btn-icon-secondary-fill-focus: var(--tds-black);
-  --tds-btn-icon-secondary-fill-active: var(--tds-black);
-  --tds-btn-icon-secondary-fill-active: var(--tds-black);
+  --tds-btn-icon-secondary-fill-active: var(--tds-grey-958);
```
Comment (nathalielindqvist) [inline packages/core/src/components/button/button-vars.scss:48]: On the Primary button, in dark mode and in active (pressed) state, there is an outline with Blue 700 while the rest of the Button is Blue 300. In Figma there is no outline at all, only the Blue 300 background color. Should this be changed?

In Figma
image


In the PR:
image

**Comment by TatyanaBumazhkina** [comment]
Checked  ghost, secondary and danger states in light/dark modes - lgtm.
Also checked comment from Natahlie about Primary/Secondary button in dark mod, but this looks like not a part of this PR

File: color/_tokens.scss [hunk 2]
```diff
@@ -1,5 +1,17 @@
 @import './mixins';
 
+$black: (
```
Comment (EmelieLitwin) [inline color/_tokens.scss:3]: I think these token names should be synced with design

**Review by alfredskeden — DISMISSED** [review]
New comments check above ^

File: packages/core/src/components/button/button-vars.scss [hunk 3]
```diff
@@ -27,25 +27,24 @@
   //Secondary
```
[the remark this answers is not in the document]
Comment (theJohnnyMe) [reply packages/core/src/components/button/button-vars.scss:27]: Good catch, I will update it!

File: packages/core/src/components/button/button-vars.scss (hunk 1 above)
Comment (theJohnnyMe) [reply packages/core/src/components/button/button-vars.scss:48]: Same as one before, it is too many issues under packages/core/dist/collection/components/button/button.css and this is just one of them. Issues needs its own ticket and time to fix.

File: packages/core/src/components/button/button-vars.scss [hunk 4]
```diff
@@ -65,17 +63,17 @@
...
+  --tds-btn-ghost-border-color-hover: var(--tds-black-48);
+  --tds-btn-ghost-outline-color-hover: var(--tds-black-48);
   --tds-btn-ghost-background-active: transparent;
   --tds-btn-ghost-color-active: var(--tds-black);
-  --tds-btn-ghost-border-color-active: rgb(0 0 0 / 87%);
+  --tds-btn-ghost-border-color-active: var(--tds-black-87);
   --tds-btn-ghost-background-focus: transparent;
   --tds-btn-ghost-color-focus: var(--tds-black);
   --tds-btn-ghost-border-color-focus: var(--tds-blue-300);
   --tds-btn-ghost-outline-color-focus: var(--tds-blue-300);
```
Comment (nathalielindqvist) [inline packages/core/src/components/button/button-vars.scss:74]: In Figma the outline color for the ghost button when focused is Blue 400 but here it is Blue 300. Which is correct?
image

Comment (theJohnnyMe) [reply packages/core/src/components/button/button-vars.scss]: Nice catch, I have updated it as per Figma file ;)

**Review by nathalielindqvist — APPROVED** [review]
LGTM!

**Review by timrombergjakobsson — APPROVED** [review]
Go for it!

**Pull request merged**